core/node/node_framework/src/implementations/layers/metadata_calculator.rs

```diff
@@ -93,3 +132,24 @@ impl Task for MetadataCalculatorTask {
         result
     }
 }
+
+#[derive(Debug)]
+pub struct TreeApiTask {
+    bind_addr: SocketAddr,
+    tree_reader: LazyAsyncTreeReader,
+}
+
+#[async_trait::async_trait]
+impl Task for TreeApiTask {
+    fn name(&self) -> &'static str {
+        "tree_api"
+    }
+
+    async fn run(self: Box<Self>, stop_receiver: StopReceiver) -> anyhow::Result<()> {
+        self.tree_reader
+            .wait()
```

Comment:
For additional responsiveness, it could make sense to select between this future and `stop_receiver`; `tree_reader.wait()` never resolves if the metadata calculator is dropped. Or you can change `tree_reader.wait()` signature to make it fallible.